x/concentrated-liquidity/swapstrategy/swap_strategy.go

```diff
@@ -104,25 +104,55 @@ func New(zeroForOne bool, sqrtPriceLimit osmomath.BigDec, storeKey sdk.StoreKey,
 // GetPriceLimit returns the price limit based on which token is being swapped in.
 // If zero in for one out, the price is decreasing. Therefore, min spot price is the limit.
 // If one in for zero out, the price is increasing. Therefore, max spot price is the limit.
-func GetPriceLimit(zeroForOne bool) osmomath.Dec {
+func GetPriceLimit(zeroForOne bool) osmomath.BigDec {
 	if zeroForOne {
-		return types.MinSpotPrice
+		return types.MinSpotPriceBigDec
 	}
-	return types.MaxSpotPrice
+	return types.MaxSpotPriceBigDec
 }
 
-func GetSqrtPriceLimit(priceLimit osmomath.Dec, zeroForOne bool) (osmomath.BigDec, error) {
+// GetSqrtPriceLimit returns sqrt price limit from price limit and swap strategy.
+// If price limit is zero and strategy is zero for one, min sqrt price is returned.
+// If price limit is zero and strategy is one for zero, max sqrt price is returned.
+// If price limit is greater than MaxSpotPrice, an error is returned.
+// Otherwise, if price limit is less that MinSpotPrice, a big decimal sqrt function
+// is used to get the sqrt price limit. Otherwise, a decimal sqrt function is used.
+// The sqrt function choice strategy applies to both zero for one and one for zero.
+// Such a choice is made to keep state-compatibility with the original at-launch
+// price range.
+func GetSqrtPriceLimit(priceLimit osmomath.BigDec, zeroForOne bool) (osmomath.BigDec, error) {
 	if priceLimit.IsZero() {
 		if zeroForOne {
-			return osmomath.BigDecFromDec(types.MinSqrtPrice), nil
+			return types.MinSqrtPriceBigDec, nil
 		}
 		return osmomath.BigDecFromDec(types.MaxSqrtPrice), nil
 	}
 
-	sqrtPriceLimit, err := osmomath.MonotonicSqrt(priceLimit)
+	if priceLimit.LT(types.MinSpotPriceV2) || priceLimit.GT(types.MaxSpotPriceBigDec) {
```

Comment:
Trying to reconcile this with the "swaps to MinInitializedTickV2 are not enabled anymore" comment above in swap tests section – should this then be checking against regular min spot price (`MinSpotPriceBigDec`) or have I misunderstood the issue?

Reply:
This was more of an added guard check for the future. It is guaranteed by the caller that `priceLimit` is above `types.MinSpotPriceBigDec`. As a result, state-compatibility is ensured

I made the check to adhere to v2 minimum right away to avoid refactoring the tests in the future PRs in the sequence.

Realize now that should have avoided doing that for clarity in the review. Will take a step by step approach in the future